smoke/IotEdgeQuickstart/details/Details.cs

```diff
@@ -334,6 +339,64 @@ protected async Task VerifyDataOnIoTHub(string moduleId)
             await eventHubClient.CloseAsync();
         }
 
+        protected async Task VerifyTwinAsync()
+        {
+            if (this.TwinTestFileName.HasValue)
+            {
+                string twinTestJson = JObject.Parse(File.ReadAllText(this.TwinTestFileName.GetOrElse(string.Empty))).ToString();
+
+                var twinTest = JsonConvert.DeserializeObject<TwinTestConfiguration>(twinTestJson);
+
+                Twin currentTwin = await this.context.RegistryManager.GetTwinAsync(this.context.Device.Id, twinTest.ModuleId);
+
+                if (twinTest.TwinTest.ContainsKey("desired") && twinTest.TwinTest["desired"].Count > 0)
+                {
+                    StringBuilder twinpatch = new StringBuilder(@"{ properties: { desired: { ");
+                    foreach (var desiredProperty in twinTest.TwinTest["desired"])
+                    {
+                        twinpatch.Append($@"{desiredProperty.Key} : {desiredProperty.Value},");
+                    }
+
+                    twinpatch.Append(@"} } }");
+
+                    // Build Patch Object.
+                    await this.context.RegistryManager.UpdateTwinAsync(this.context.Device.Id, twinTest.ModuleId, twinpatch.ToString(), currentTwin.ETag);
+                }
+
+                if (twinTest.TwinTest.ContainsKey("expectedReported") && twinTest.TwinTest["expectedReported"].Count > 0)
+                {
+                    using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(20)))
+                    {
+                        bool foundAtLeastOne = false;
+                        while (true)
+                        {
+                            await Task.Delay(TimeSpan.FromSeconds(10), cts.Token);
```

Comment:
why need a delay here?

Reply:
So, we don't keep pulling right after we set. It will take some time (usually couple of seconds) before the module receives the desired property and update reported property.

Reply:
Not Applicable anymore, since I'm using retry.do.